js/bitget3.js

```diff
@@ -0,0 +1,2899 @@
+        //         "errMsg":null,
+        //         "data":[
+        //             {"id":"1594694700000","amount":"283.6811","count":"234","open":"9230.00","close":"9227.15","low":"9206.66","high":"9232.33","vol":"2618015.032504000000"},
+        //             {"id":"1594695600000","amount":"457.2904","count":"238","open":"9227.15","close":"9229.46","low":"9223.80","high":"9235.14","vol":"4220734.684570000000"},
+        //             {"id":"1594696500000","amount":"501.2353","count":"255","open":"9229.46","close":"9227.78","low":"9222.69","high":"9230.74","vol":"4625779.185006000000"},
+        //         ]
+        //     }
+        //
+        // swap
+        //
+        //     [
+        //         ["1594764900000","9255.5","9261","9251","9255.5","3958946","427.742307964305"],
+        //         ["1594765800000","9255.5","9264","9252","9258","3609496","389.832756058107"],
+        //         ["1594766700000","9258","9260","9244.5","9250.5","3738600","403.97870345085"],
+        //     ]
+        //
+        let candles = response;
+        if (!Array.isArray (response)) {
+            candles = this.safeValue (response, 'data', []);
+        }
+        return this.parseOHLCVs (candles, market, timeframe, since, limit);
+    }
+
+    parseSpotBalance (response) {
+        //
+        //     {
+        //         "status":"ok",
+        //         "ts":1595681450932,
+        //         "data":{
+        //             "list":[
+        //                 {"balance":"0.0000000000000000","currency":"BTC","type":"trade"},
+        //                 {"balance":"0.0000000000000000","currency":"BTC","type":"frozen"},
+        //                 {"balance":"0.0000000000000000","currency":"BTC","type":"lock"},
+        //             ],
+        //             "id":"7420922606",
+        //             "type":"spot",
+        //             "state":"working"
+        //         }
+        //     }
+        //
+        const result = { 'info': response };
+        const data = this.safeValue (response, 'data');
+        const balances = this.safeValue (data, 'list');
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            if (!(code in result)) {
+                const account = this.account ();
+                result[code] = account;
+            }
+            const type = this.safeValue (balance, 'type');
+            if (type === 'trade') {
+                result[code]['free'] = this.safeString (balance, 'balance');
+            } else if ((type === 'frozen') || (type === 'lock')) {
+                const used = this.safeString (result[code], 'used');
+                result[code]['used'] = Precise.stringAdd (used, this.safeString (balance, 'balance'));
+            }
+        }
+        return this.parseBalance (result);
```

Comment:
`parseBalance` is not an inherited method and as does not exist in this class crashes the code. Probably you meant `safeBalance` right?

Reply:
Fixed in PR #11907